src/navparser.cpp

```diff
@@ -95,24 +97,31 @@ float getZBetweenAreas(CNavArea *start, CNavArea *end)
 static std::unordered_map<std::pair<CNavArea *, CNavArea *>, ignoredata, boost::hash<std::pair<CNavArea *, CNavArea *>>> ignores;
 namespace ignoremanager
 {
-static bool vischeck(CNavArea *begin, CNavArea *end)
+static ignore_status vischeck(CNavArea *begin, CNavArea *end)
 {
     Vector first  = begin->m_center;
     Vector second = end->m_center;
-    first.z += 42;
-    second.z += 42;
-    return IsVectorVisible(first, second, true, LOCAL_E, MASK_PLAYERSOLID);
+    first.z += 70;
+    second.z += 70;
+    ignore_status status = vischeck_failed;
+    // Is world blocking it?
+    if (IsVectorVisibleNavigation(first, second, LOCAL_E, MASK_PLAYERSOLID))
+    {
+        status = vischeck_success;
+        // Is something else blocking it?
+        if (!IsVectorVisible(first, second, true, LOCAL_E, MASK_PLAYERSOLID))
+            status = vischeck_blockedentity;
+    }
+    return status;
```

Comment:
What even is this status nonsense?
```suggestion
    // Is world blocking it?
    if (IsVectorVisibleNavigation(first, second, LOCAL_E, MASK_PLAYERSOLID))
    {
        // Is something else blocking it?
        if (!IsVectorVisible(first, second, true, LOCAL_E, MASK_PLAYERSOLID))
            return vischeck_blockedentity;
        else
            return vischeck_success;
    }
    return vischeck_failed;
```

Reply:
i think my brain shut down there